build/parsePreamble.cc

```diff
@@ -320,10 +321,15 @@ int addSource(rpmSpec spec, int specline, const char *srcname, rpmTagVal tag)
 	return RPMRC_FAIL;
     }
 
-    /* Create the entry and link it in */
+    /* Create the entry and link it at the end */
     p = newSource(num, srcname, flag);
-    p->next = spec->sources;
-    spec->sources = p;
+    if (!spec->sources) {
+	spec->sources = p;
+    } else {
+	Source * s;
+	for (s = spec->sources; s->next; s=s->next);
```

Comment:
I'd find this more obvious:
```
Source * s = spec->sources;
while (s->next)
    s= s->next;
s->next = p;
```

but if you want to use for, at least put {} at the end to make it obvious this is an intentionally empty for-loop.